### PR TITLE
ref(makeTestQueryClient): Use makeTestQueryClient consistently in tests

### PR DESCRIPTION
**Pull request by ryan953**
See also: https://github.com/getsentry/develop/pull/1036

### CODE REVIEW

File: static/app/utils/profiling/hooks/useProfileFunctions.spec.tsx [hunk 1]
```diff
@@ -1,19 +1,18 @@
...
 
 import {useProfileFunctions} from 'sentry/utils/profiling/hooks/useProfileFunctions';
-import {QueryClient, QueryClientProvider} from 'sentry/utils/queryClient';
+import {QueryClientProvider} from 'sentry/utils/queryClient';
 import {OrganizationContext} from 'sentry/views/organizationContext';
 
 function TestContext({children}: {children: ReactElement}) {
   const {organization} = useMemo(() => initializeOrg(), []);
-  // ensure client is rebuilt on each render otherwise caching will interfere with subsequent tests
-  const client = useMemo(() => new QueryClient(), []);
```
Comment (ryan953) [inline static/app/utils/profiling/hooks/useProfileFunctions.spec.tsx:13]: `makeTestQueryClient` explicitly handles this

File: static/app/views/discover/table/quickContext/eventContext.spec.tsx [hunk 2]
```diff
@@ -40,18 +41,10 @@ const dataRow: EventData = {
...
 };
 
-const queryClient = new QueryClient({
-  defaultOptions: {
-    queries: {
-      retry: false,
-    },
-  },
-});
-
```
Comment (ryan953) [inline static/app/views/discover/table/quickContext/eventContext.spec.tsx:50]: `makeTestQueryClient` has the same defaults

**Review by Zylphrex — APPROVED** [review]

**Review by scttcper — APPROVED** [review]

**Comment by JonasBa** [comment]
@ryan953 can we do something about linters here so folks don't end up importing QueryClient in test files (it feels like it's bound to happen :D)

**Comment by ryan953** [comment]
> @ryan953 can we do something about linters here so folks don't end up importing QueryClient in test files (it feels like it's bound to happen :D)

Ya we should do something. Maybe we need to followup and put `makeTestQueryClient` into the mocks folder so that jest picks the mock every time. That way it's written the same everywhere. Not sure why it's not a mock already tbh.

Another thing we could do is put together a helper for testing hooks, we have `render` for testing components which sets up a lot of context like this, but for hooks there's nothing. Either a helper, or i wonder if RTL has a way to register default context globally so that, again, people can keep using `reactHooks.renderHook` and not have to be told.

**Pull request merged**